src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -249,7 +249,12 @@ private void SetUrl()
                 }
 
                 EventSourceTrace("Url: {0}", requestUri);
-                SetCurlOption(CURLoption.CURLOPT_URL, requestUri.AbsoluteUri);
+                string idnHost = requestUri.IdnHost;
```

Comment:
The results are not being cached in IdnHost, and there were extra argument checks in DnsSafeHost whose results are cached, and then there's an extra condition check on DnsSafeHost result. Not sure how to measure impact.